Library/Homebrew/cask/dsl.rb

```diff
@@ -333,6 +333,20 @@ def deprecated?
       @deprecated == true || @caveats&.discontinued? == true
     end
 
+    # TODO: replace with with attr_reader once discontinued? is disabled
+    def deprecation_date
+      return @deprecation_date if @deprecation_date.present?
+
+      Date.today if @caveats&.discontinued? == true
```

Comment:
This seems like it will behave inconsistently, e.g. if it's the 17th on the API generation machine but the 18th on the end user.

Do we even need `caveats { discontinued }` to map to `depreacated?` if we're going to have RuboCop complain about it and be swapping all usages in a couple days?

Reply:
Yeah maybe not, but until 4.2.0 is tagged and the PRs are merged, all discontinued casks are suddenly regular if you're using the API. Maybe we don't care since the only major logic that depends on that right now is an install warning, `brew livecheck`, and `brew audit`.

Thoughts?

Reply:
In theory, third parties could have specified `discontinued` and not be using `brew style` so this could cause issues for them since it's not officially deprecated until 4.3.0.

Reply:
To fix the timing issue, thoughts on just returning `2023-12-17` (today)?

Reply:
> Yeah maybe not, but until 4.2.0 is tagged and the PRs are merged, all discontinued casks are suddenly regular if you're using the API.

Is this any different to how it was yesterday? Seems like it will behave exactly like it always has for API use cases.

> brew livecheck, and brew audit

These are probably fine if we have the cop.

> In theory, third parties could have specified discontinued and not be using brew style so this could cause issues for them since it's not officially deprecated until 4.3.0.

Sounds like everything will behave as it did before except `brew livecheck` and `brew audit`. We could add a `cask.discontinued?` check for those two until 4.3.0, though the latter probably matters less as it auto-runs `brew style` unless you do `--skip-style`

Reply:
Oh yeah you're right—the caveat will still show in those cases

I'll update this